ds3_java_cli/src/main/java/com/spectralogic/ds3cli/command/GetBlobsOnTape.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * ******************************************************************************
+ *   Copyright 2016 Spectra Logic Corporation. All Rights Reserved.
+ *   Licensed under the Apache License, Version 2.0 (the "License"). You may not use
+ *   this file except in compliance with the License. A copy of the License is located at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   or in the "license" file accompanying this file.
+ *   This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ *   CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ *   specific language governing permissions and limitations under the License.
+ * ****************************************************************************
+ */
+
+package com.spectralogic.ds3cli.command;
+
+import com.spectralogic.ds3cli.Arguments;
+import com.spectralogic.ds3cli.exceptions.CommandException;
+import com.spectralogic.ds3cli.models.GetBlobsOnTapeResult;
+import com.spectralogic.ds3cli.util.Ds3Provider;
+import com.spectralogic.ds3cli.util.FileUtils;
+import com.spectralogic.ds3client.commands.spectrads3.*;
+import com.spectralogic.ds3client.networking.FailedRequestException;
+import com.spectralogic.ds3client.utils.Guard;
+import com.spectralogic.ds3client.utils.SSLSetupException;
+import org.apache.commons.cli.MissingOptionException;
+
+import java.io.IOException;
+import java.security.SignatureException;
+import java.util.UUID;
+import static com.spectralogic.ds3cli.util.Utils.nullGuardToString;
+
+
+public class GetBlobsOnTape extends CliCommand<GetBlobsOnTapeResult> {
+
+    // Barcode or tape ID
+    private String tapeId;
+    private String barcode;
+
+    public GetBlobsOnTape(final Ds3Provider provider, final FileUtils fileUtils) {
+        super(provider, fileUtils);
+    }
+
+    @Override
+    public CliCommand init(final Arguments args) throws Exception {
+        this.tapeId = args.getId();
+        if (Guard.isStringNullOrEmpty(this.tapeId)) {
+            throw new MissingOptionException("The get_blobs_on_tape command requires '-i' to be set with the tape Id or barcode");
+        }
+        // pass in a string, could be UUID or Barcode
+        try {
+            UUID testId = (UUID.fromString(this.tapeId));
+            this.barcode = "";
+        } catch (IllegalArgumentException ex) {
+            // not a UUID
+            this.barcode = this.tapeId;
+            this.tapeId = "";
+        }
+        return this;
+    }
+
+    @Override
+    public GetBlobsOnTapeResult  call() throws IOException, SignatureException, SSLSetupException, CommandException {
```

Comment:
The logic here can be simplified to a single api call.  By issuing GetBlobsOnTape, if the client throughs a request failed exception with a status code of 404, then we know that the tape in question does not exist.  By doing that we also don't have to check to see what the format of the barcode or tapeId is.  We just pass it along to the  sdk and let the api handle it.